salesagility/SuiteCRM
# Fix #9691 - Add PDF template functionality to more modules automatically

## Description
This PR implements the following features to add the functionality of PDF Templates automatically in the modules displayed in the CRM menu.

1. Add a function in the PDF Templates module that lists the modules added in the CRM menu. (modules/AOS_PDF_Templates/AOS_PDF_Templates.php)

2. Eliminate the default values ​​from the pdf_template_type_dom list, associated list with the Type field of the PDF Templates module, and assign it the values ​​returned by the previous function. (modules/AOS_PDF_Templates/AOS_PDF_Templates.php and include/language/en_us.lang.php)

3. Define two exclusion arrays:
3.1 $excludedModulesForCreateTemplates -> Modules that will not be shown in the Type field of a PDF Template.
3.2 $excludedModulesToAddButtons -> Modules where the PDF print button will not be shown, neither in the list view nor in the detail view. The AOS_Contracts, AOS_Quotes and AOS_Invoices modules are excluded for having the display of this custom button.
(modules/AOS_PDF_Templates/AOS_PDF_Templates.php)

4. Eliminate the code that added the functionality to the Contacts, Accounts and Leads modules since it is not customized and will be correctly replaced by the logic that we indicate in the next point. (modified files in modules/Accounts, modules/Contacts and modules/Leads )

5. Add the functionality of PDF Templates in the non-excluded modules, included in custom modules already created or when they are created, both in the list view and in the detail view. (modules/Notes/metadata/detailviewdefs.php and include/DetailView/DetailView2.php and include/ListView/ListViewDisplay.php)

6. Add the EDIT, DUPLICATE and DELETE actions in the Notes detail view and in the modules creation process, in the file type module template. Now, both the Notes module and the template need the action's array to be empty to load the default actions, a fact that will not be fulfilled since we automatically include the action for the generation of PDF templates (include/SugarObjects/templates/file/metadata/detailviewdefs .php)

## Motivation and Context
Have the functionality of PDF templates in the active modules in the CRM

## How To Test This
1. Go to the PDF templates module and access the editing view of a template.
2. Check that the Type field shows the modules that appear in the CRM menu
3. Create several templates and check that it works in the corresponding modules
4. Check that the EDIT, DUPLICATE and DELETE actions are added in all modules, especially in Notes and in modules created from the creation template for file-type modules
5. Check that the AOS_Contracts, AOS_Quotes and AOS_Invoices modules maintain their custom behavior

## Types of changes


### Final checklist
